readthedocs/subscriptions/views.py

```diff
@@ -103,10 +109,28 @@ def get_object(self):
         We retry the operation when the user visits the subscription page.
         """
         org = self.get_organization()
-        return (
-            Subscription.objects
-            .get_or_create_default_subscription(org)
-        )
+        return org.stripe_subscription
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        stripe_subscription = self.get_object()
+        if stripe_subscription:
+            context[
+                "features"
+            ] = self.get_organization().subscription.plan.features.all()
+
+            stripe_price = stripe_subscription.items.first().price
+            context["stripe_price"] = stripe_price
+
+            # When Stripe marks the subscription as ``past_due``,
+            # it means the usage of RTD service for the current period/month was not paid at all.
+            # show the end date as the last period the customer paid.
+            context["subscription_end_date"] = stripe_subscription.current_period_end
+            if stripe_subscription.status == SubscriptionStatus.past_due:
+                context[
+                    "subscription_end_date"
+                ] = stripe_subscription.current_period_start
```

Comment:
Copied this from https://github.com/readthedocs/readthedocs.org/blob/d640a0db3a677df46c3004228f7f76e188f8451f/readthedocs/subscriptions/managers.py#L163-L170, does it make sense to follow that logic or maybe just always show `current_period_end`?

Reply:
```
In [4]: Subscription.objects.filter(status='past_due').count()
Out[4]: 6
```

We have only 6 of these, and all of them are just one invoice behind. But it makes sense to show the data from the latest paid invoce.

Reply:
Ohh, that's probably because we are being more aggressive now and canceling the subscription after 2 months without a valid payment. So, we won't have too many cases for this probably, but we will have some. 👍🏼 to showing the data from latest payment, then.